### DISCUSSION TITLE
How to have `:meth:` references with `~` show the type name.

### DISCUSSION THREAD

**Discussion by ktbarrett** [sphinx-doc/sphinx#12548] [Q&A] [answered]
Standard references (``:meth:`cocotb_tools.runner.Runner.test` ``) show fully qualified. Adding the tilde character at the beginning (``:meth:`~cocotb.runner.Runner.test` ``) shows just the last part. But often what I want to do when referencing methods *out-of-context from the class where it is defined* is to show the class name and the method name, as showing only the method name is not useful. I end up having to write the following: ``:meth:`Runner.test() <cocotb.runner.Runner.test>` ``. This is kinda verbose.

Can there be another special character introduced to semantically valuable reference shortenings rather than just "last part" like `~`? Is something like this built in already as a way to configure the `~` operator?

**Comment by AA-Turner** [comment, marked as answer]
I would suggest ``:meth:`.Runner.test` ``, see the paragraphs at the end of [cross-referencing Python objects](https://www.sphinx-doc.org/en/master/usage/domains/python.html#cross-referencing-python-objects). Note that if you have multiple `Runner.test` methods in different modules, this will raise a warning -- instead you could make use of the `.. currentmodule::` directive.

A
